core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -141,7 +140,14 @@ public void onNext(Frame frame) {
                 .pollInterval(DynamicPollInterval.ofMillis(50))
                 .pollInSameThread()
                 .until(
-                    () -> client.inspectContainerCmd(ryukContainerId).exec(),
+                    () -> {
+	                        try {
+	                            return client.inspectContainerCmd(ryukContainerId).exec();
+	                        } catch(NotFoundException e) {
+	                            log.debug("Ryuk container cannot be found and probably had a problem starting. Ryuk's logs:\n{}", ryukLog);
```

Comment:
note that `ryukLog` will grow instead of being "per attempt", so perhaps we should only log it when `until` fails after all attempts

Reply:
Hi @bsideup , should I surround the whole `ContainerState containerState = new ContainerState() { ... }` block with the try/catch?

Reply:
Yes, I think that should work fine.